### PR TITLE
Add new version for Microsoft.Solutions

### PR DESCRIPTION
**Pull request by dandanwang0320**
Update to the latest api version.
Move Jit request changes to this PR.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 1]
```diff
@@ -0,0 +1,1949 @@
...
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "put": {
+        "tags": [
+          "Applications"
+        ],
+        "operationId": "Applications_CreateOrUpdate",
+        "description": "Creates a new managed application.",
```
Comment (yungezz) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Create or Update per operationId?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 2]
```diff
@@ -0,0 +1,1949 @@
...
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/applications": {
+      "get": {
+        "tags": [
+          "Applications"
+        ],
+        "operationId": "Applications_ListByResourceGroup",
+        "description": "Gets all the applications within a resource group.",
```
Comment (yungezz) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: List all...

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 3]
```diff
@@ -0,0 +1,1949 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{applicationId}": {
```
Comment (yungezz) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: /subscriptions/{subscriptionId}/providers/Microsoft.Solutions/applications/{applicationId} ?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 4]
```diff
@@ -0,0 +1,1949 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the managed application.",
+            "schema": {
+              "$ref": "#/definitions/Application"
+            }
+          },
+          "404": {
```
Comment (yungezz) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: normally error code not in swagger

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 5]
```diff
@@ -0,0 +1,1949 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
```
Comment (ryansbenson) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Async actions should return 202: 
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#call-action-post

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 6]
```diff
@@ -0,0 +1,1949 @@
...
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to use for getting the next set of results."
+        }
+      },
+      "description": "List of managed applications."
+    },
+    "JitRequestProperties": {
+      "properties": {
+        "ApplicationResourceId": {
```
Comment (ryansbenson) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Is the intent to have a LinkedAccessCheck on this field?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 7]
```diff
@@ -0,0 +1,1949 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "patch": {
```
Comment (ryansbenson) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Patch should not take in the same body model as Put. Either introduce a new type definition containing the specific properties that are patchable or remove the Patch API.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 8]
```diff
@@ -0,0 +1,1949 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (ryansbenson) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: If you keep Patch, 200 represents a synchronous response (no need for x-ms-long-running-operation) or if async then 201 should be returned.
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 9]
```diff
@@ -0,0 +1,1949 @@
...
+            "$ref": "#/definitions/JitRequestDefinition"
+          },
+          "description": "The array of Jit request definition."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to use for getting the next set of results."
+        }
+      },
+      "description": "List of managed applications."
```
Comment (ryansbenson) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This description should be updated.

**Comment by yungezz** [comment]
Hi @dandanwang0320  could  you pls address comments from reviewing board?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 1 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Sorry these are just copied from the old version. Could you ignore the first commit during review?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 3 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This is the fully qualified application Id. Also it's from the old version.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 2 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This is also from the old version.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 5 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: So in our code base, it's currently HttpStatusCode.Created(201). So which side should we change first? Or could we do these at the same time?

**Comment by yungezz** [comment]
Hi @dandanwang0320 are you still on this PR? or is it ok to close the PR if you're not ready for reviewing?

**Comment by dandanwang0320** [comment]
@yungezz Yes. I replied some of your comments. Synced with Bryan offline for some necessary changes needed within ARM code.

**Comment by yungezz** [comment]
HI @ryansbenson could you pls review the PR again after author's updating?

**Comment by dandanwang0320** [comment]
Will add more API/Properties as I'm fixing Ryan's comments.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 8 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Will remove long running.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 6 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Updated Microsoft.Solutions manifest. Thanks.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 5 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: https://msazure.visualstudio.com/One/_git/AzureUX-ARM/pullrequest/2210930?_a=overview PR created to change back end.

**Comment by dandanwang0320** [comment]
Hi @ryansbenson, I've resolved your comments and added several more resources to the new version to keep this updated.
Please take another look.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 10]
```diff
@@ -0,0 +1,2134 @@
...
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/jitRequests/{jitRequestName}": {
+      "get": {
+        "tags": [
+          "jitRequests"
+        ],
+        "operationId": "jitRequests_GetByName",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: "JitRequests_Get"

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 11]
```diff
@@ -0,0 +1,2134 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },          
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This should only return 201 if it is creating.  200 is also a valid response if the JITRequest already existed

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 12]
```diff
@@ -0,0 +1,2134 @@
...
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "patch": {
+        "tags": [
+          "jitRequests"
+        ],
+        "operationId": "jitRequests_Patch",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: JitRequests_Update for consistency with other RPs

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 13]
```diff
@@ -0,0 +1,2134 @@
...
+        "description": "Creates or updates the JIT request.",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: "Updates the JIT request."

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 14]
```diff
@@ -0,0 +1,2134 @@
...
+        } 
+      }     
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/jitRequests": {
+      "get": {
+        "tags": [
+          "jitRequests"
+        ],
+        "operationId": "jitRequests_ListByResourceGroup",
+        "description": "Retrieves all JIT requests within the subscription.",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: "within the resource group"

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 15]
```diff
@@ -0,0 +1,2134 @@
...
+          }
+        } 
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/applications/{applicationName}/refreshPermissions": {
+      "post": {
+        "tags": [
+          "refreshPermissions"
+        ],
+        "operationId": "refreshPermissions_Post",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: "Applications_RefreshPermissions"

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 16]
```diff
@@ -0,0 +1,2134 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        } 
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/applications/{applicationName}/refreshPermissions": {
+      "post": {
+        "tags": [
+          "refreshPermissions"
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: "Applications"

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 17]
```diff
@@ -0,0 +1,2134 @@
...
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to use for getting the next set of results."
+        }
+      },
+      "description": "List of JIT requests."
+    },
+    "JitRequestProperties": {
+      "properties": {
+        "ApplicationResourceId": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Please use camel-case for property names for consistency

Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: If the user needs to provide the application ID should jitRequests be a child of applications?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 18]
```diff
@@ -0,0 +1,2134 @@
...
+          "type": "string",
+          "description": "The parent application id."
+        },
+        "PublisherTenantId": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: publisher of what?  The user needs to provide this?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 19]
```diff
@@ -0,0 +1,2134 @@
...
+        },
+        "JitAuthorizationPolicies": {
+          "description": "The JIT authorization policies.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JitAuthorizationPolicies"
+          }
+        },
+        "JitSchedulingPolicy": {          
+          "$ref": "#/definitions/JitRequestProperties",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This is a circular reference.  Is that intended?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 20]
```diff
@@ -0,0 +1,2134 @@
...
+      },
+      "required": [
+        "PrincipalId",
+        "RoleDefinitionId"
+      ],
+      "description": "The JIT authorization policies."
+    },
+    "JitSchedulingPolicy": {
+      "properties": {
+        "Type": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: can you call this "schedulingType" or similar so it is not confused with the top-level "type" property?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 21]
```diff
@@ -0,0 +1,2134 @@
...
+      ],
+      "description": "The JIT scheduling policies."
+    },
+    "ApplicationClientDetails": {
+      "properties": {
+        "Oid": {
+          "type": "string",
+          "description": "The client Oid."
+        },
+        "Puid": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Aren't PUIDs PII?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 22]
```diff
@@ -0,0 +1,2134 @@
...
+      "enum": [
+        "NotSpecified",
+        "Once",
+        "Recurring"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "JitSchedulingType",
+        "modelAsString": true
+
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: "metadata"

Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: which of these properties do the user provide and which can be marked as readonly?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 23]
```diff
@@ -0,0 +1,2134 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the application definition artifact.",
+            "schema": {
+              "type": "string"
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This returns a string?  It should be returning a JSON collection { "value": [], "nextLink": null }

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 23 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: The schema is defined by the input. So the caller needs to deserialize the response themselves.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 20 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This type should always be referenced as JitSchedulingPolicy.Type, it should be relatively safe.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 18 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This is the publisher of the managed application(ISV). This should be in the response.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 17 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: Users calling Jit request apis don't necessarily have access to the whole application.
We've added a linked access check in the JitRequest manifest.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 21 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: I've discussed internally, this should be fine.

**Comment by dandanwang0320** [comment]
@pilor I've responded/fixed the comments you left. We decided to remove one of the APIs since it's facing internal customers only(updateAccess).

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json (hunk 18 above)
Comment (pilor) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: This should be marked as readOnly if it is only in the response

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 24]
```diff
@@ -0,0 +1,2004 @@
...
+          "readOnly": true,
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The JIT request provisioning state."
+        },
+        "jitRequestState": {
+          "readOnly": true,
+          "$ref": "#/definitions/JitRequestState",
+          "description": "The JIT request state."
+        },
+        "createdBy": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: createdBy and updatedBy should be readOnly

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json [hunk 25]
```diff
@@ -0,0 +1,2004 @@
...
+      "description": "List of JIT requests."
+    },
+    "JitRequestProperties": {
+      "properties": {
+        "applicationResourceId": {
+          "type": "string",
+          "description": "The parent application id."
+        },
+        "publisherTenantId": {
+          "type": "string",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json]: publisherTenantId should be readOnly

**Review by pilor — CHANGES_REQUESTED** [review]

**Review by pilor — APPROVED** [review]

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
@dandanwang0320 is this PR ready to merge?

**Comment by dandanwang0320** [comment]
> @dandanwang0320 is this PR ready to merge?

Let's do it next Monday. Is it Ok?

**Pull request merged**